### PR TITLE
[BUGFIX release] Don’t leak container while injection deprecated cont…

### PR DESCRIPTION
**Pull request by stefanpenner**
…ainer

### CODE REVIEW

**Comment by rwjblue** [comment]
The approach looks good, but there is some sort of failure still...

File: packages/container/lib/container.js [hunk 1]
```diff
@@ -527,25 +517,27 @@ function factoryInjectionsFor(container, fullName) {
...
-    }
-  });
+    return value;
+  }
+};
+
+// TODO - remove when Ember reaches v3.0.0
+function injectDeprecatedContainer(object) {
+  if ('container' in object) { return; }
+  Object.defineProperty(object, 'container', INJECTED_DEPRECATED_CONTAINER_DESC);
```
Comment (stefanpenner) [inline packages/container/lib/container.js]: I need to actually set the container again here `setOwner(object, container)`

Comment (stefanpenner) [inline packages/container/lib/container.js]: looks like `container.owner` is null for the offending tests. and this PR closing over `container` masked that.

@rwjblue do i expect `container.owner` to be null? Is that a test config error?

Comment (stefanpenner) [inline packages/container/lib/container.js]: I'll actually investigate more, just gotta head out now.

**Comment by stefanpenner** [comment]
@rwjblue / @krisselden fixed, ready for review

**Review by rwjblue — APPROVED** [review]

**Review by krisselden — COMMENTED** [review]
Why are we setting override instead of setOwner? Also, the owner is supposed to be the app instance and likely that test didn't have an app instance.  Object in the above is the prototype with no double extend so this would still leak

**Comment by stefanpenner** [comment]
Because the original code was explicitly returning the container (Not whatever the owner happened to be) I didn't want to change behavior for a leak fix.

I think as a next step we should likely just remove this code entirely in canary, thoughts?

**Comment by stefanpenner** [comment]
Or a better question, what would need to be done to remove this in canary? It may just be update the deprecation message in release/beta

**Pull request merged**